en/docs/extended_topics/implement_data_loader.md

```diff
@@ -0,0 +1,461 @@
+# Customize DataLoader
+
+As described in [Data Input](../basics_topics/data_input.md), OneFlow supports two ways to load data: one is directly use Numpy data, the other one is use DataLoader and some relative operators. 
+
+Under the large industrial scene, Data Loading can easily become the bottleneck through the training process. Since we use DataLoader and some preprocessing operators, OneFlow's acceleration mechanism helps to load and preprocess data more efficiently, which can solve the problem. 
+
+To use DataLoader in OneFlow, we usually apply `XXXReader` to load the file data, and we use `XXXDecode` to decode or preprocess the data. These two operators work together to complete the function of DataLoader.  
+
+Now OneFlow has built some data format's [DataLoader](https://oneflow.readthedocs.io/en/master/search.html?q=reader&check_keywords=yes&area=default) internally. If we want to use DataLoader to promote the efficiency of data loading, however, the DataLoader for the corresponding data format is not yet built in OneFlow. At this time, we can implement our own DataLoader to load the customized data format.  
+
+In this article we implement a Mini Dataloader. You can check the [Complete Code](https://github.com/Oneflow-Inc/oneflow-documentation/tree/master/en/docs/code/extended_topics/data_loader/) in this repository. 
+
+As an example, the data format that Mini Dataloader supported is : A plain text file with two columns of numbers separated by commas (See the `part-000` and `part-001` file in code): 
+
+```text
+1.01,2.02
+2.01,4.02
+3.0,6.05
+4.1,8.205
+5,10
+6.0,12.0
+7.0,14.2
+8.0,16.3
+9.1,18.03
+```
+
+This article will take Mini Dataloader as an example to explain the key points of implementing customized DataLoader. 
+
+# The composition of Dataloader
+
+A complete Dataloader generally includes two types of Op: 
+
+- Data Reader: Which is responsible for loading the data in file system to the input stream of memory and setting the data to the Op's output. 
+- Data Decoder: The Data Decoder decodes and output the data in Data Reader Op. 
+
+For some simple data formats, which is no need for decoding. We can omit the Data Decoder and just use Data Reader. 
+
+As an example, since the data format processed by Mini Dataloader is simple. We still implement the Data Reader and Data Decoder. Among these two Ops: 
```

Comment:
As an example, though the data format processed by Mini Dataloader is simple, we still implement the two types of ops: Data Reader and Data Decoder.